### PR TITLE
Bug Fixes

### PR DESCRIPTION
**Pull request by bardielle**

### CODE REVIEW

File: examples/machine-with-user-data.yaml [hunk 1]
```diff
@@ -14,14 +13,9 @@ spec:
       node-role.kubernetes.io/infra: ""
   providerSpec:
     value:
+      # apiVersion: awsproviderconfig.openshift.io/v1beta1
```
Comment (chenyosef) [inline examples/machine-with-user-data.yaml:16]: awsproviderconfig ?

File: pkg/client/client.go [hunk 2]
```diff
@@ -102,9 +101,9 @@ func (c *kubevirtClient) DeleteVirtualMachine(namespace string, name string, opt
 	return c.kubecliclient.VirtualMachine(namespace).Delete(name, options)
 }
 
-func (c *kubevirtClient) GetVirtualMachine(namespace string, name string) (*v1.VirtualMachine, error) {
+func (c *kubevirtClient) GetVirtualMachine(namespace string, name string, options *k8smetav1.GetOptions) (*v1.VirtualMachine, error) {
 	// options *k8smetav1.GetOptions would be nil
-	return c.kubecliclient.VirtualMachine(namespace).Get(name, nil)
```
Comment (chenyosef) [inline pkg/client/client.go:107]: Remove also the comment above - no longer relevant

File: pkg/managers/vm/machine_scope.go [hunk 3]
```diff
@@ -170,22 +178,78 @@ func (s *machineScope) setMachineCloudProviderSpecifics(vm *kubevirtapiv1.Virtua
...
+				DataVolume: &kubevirtapiv1.DataVolumeSource{
+					Name: virtualMachineName + "-bootvolume" + "",
+				},
+			},
+		},
+	}
+
+	template.Spec.Domain = kubevirtapiv1.DomainSpec{}
+
+	requests := corev1.ResourceList{}
```
[the remark this answers is not in the document]
Comment (bardielle) [reply pkg/managers/vm/machine_scope.go:203]: It's not - Im assigned it in line 205 + 207

File: pkg/managers/vm/machine_scope.go [hunk 4]
```diff
@@ -170,22 +178,78 @@ func (s *machineScope) setMachineCloudProviderSpecifics(vm *kubevirtapiv1.Virtua
...
+	//Memory:
+	requests[corev1.ResourceMemory] = apiresource.MustParse("2048M")
+
+	//CPU:
+	//requests[corev1.ResourceCPU] = apiresource.MustParse("2")
+
+	template.Spec.Domain.Resources = kubevirtapiv1.ResourceRequirements{
+		Requests: requests,
+	}
+	template.Spec.Domain.Machine = kubevirtapiv1.Machine{Type: ""}
```
[the remark this answers is not in the document]
Comment (bardielle) [reply pkg/managers/vm/machine_scope.go:213]: I copied it from Ravid's VM yaml 
@ravidbro - What that type means?

File: examples/machine-with-user-data.yaml [hunk 5]
```diff
@@ -14,14 +13,9 @@ spec:
...
-      SecretName:
-        name: underkube-kubeconfig
-      SourcePvcName:
-        name: my-image
-
-  versions:
-    kubelet: 1.10.1
-    controlPlane: 1.10.1
+      SecretName: underkube-config
+      SourcePvcName: my-image
```
[the remark this answers is not in the document]
Comment (bardielle) [reply examples/machine-with-user-data.yaml:21]: Changed it to pvc-rhcos-image

File: pkg/managers/vm/machine_scope.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (ravidbro) [reply pkg/managers/vm/machine_scope.go:217]: I think I just copied an example from the kubevirt repo

[the remark this answers is not in the document]
Comment (chenyosef) [reply pkg/managers/vm/machine_scope.go:217]: The only mahcine type supported by kubevirt is "q35" (while the underlying libvirt supports many more - so I guess it either need to be set to q35 (and not based on user input), or remove all together (in that case I believe kubevirt will use the default q35)
https://github.com/kubevirt/kubevirt/issues/1230

**Pull request merged**